api/backup.go

```diff
@@ -117,18 +117,25 @@ func (s *Server) BackupDelete(w http.ResponseWriter, req *http.Request) error {
 		return err
 	}
 	if input.Name == "" {
-		return errors.Errorf("empty backup name is not allowed")
+		return errors.New("empty backup name is not allowed")
 	}
 
 	volName := mux.Vars(req)["volName"]
 
-	if err := s.m.DeleteBackup(input.Name, volName); err != nil {
-		return errors.Wrapf(err, "error deleting backup %v of volume %v", input.Name, volName)
+	backup, err := s.m.GetBackup(input.Name, volName)
+	if err != nil {
+		logrus.WithError(err).Warnf("failed to get backup '%v' of volume '%v'", input.Name, volName)
+	}
```

Comment:
The `backup` from `GetBackup` is unused. Why not directly do `DeleteBackup`?

Reply:
The 'GetBackup' will return error if API exception to get backup or the backup is not exists. 

Therefore, I log the error but checking if 'backup' is null or not and perform deletion if still exists.

If I execute 'DeleteBackup' directly, it will be the same as the original situation, causing volumesnapshot cannot be removed.

Reply:
I see. The code flow is a bit of confusing.
Will DeleteBackup return NOTEXIST error? We can ignore the error and continue with the remaining tasks if yes. Or, suggest adding a comment here.

Reply:
Yes, the error message will mention that the backup does not exist.

But the current way of judging is to parse the error string, because there is no error code or HTTP status to distinguish the error category.

I will add comment to describe the context and also create a ticket to see if the API could be enhanced to export the HTTP status code.